core/commonJs/src/internal/Platform.kt

```diff
@@ -0,0 +1,153 @@
+    /**
+     * References:
+     * - https://github.com/js-joda/js-joda/blob/8c1a7448db92ca014417346049fb64b55f7b1ac1/packages/timezone/src/MomentZoneRulesProvider.js#L78-L94
+     * - https://github.com/js-joda/js-joda/blob/8c1a7448db92ca014417346049fb64b55f7b1ac1/packages/timezone/src/unpack.js
+     * - <https://momentjs.com/timezone/docs/#/zone-object/>
+     */
+    fun unpackBase60(string: String): Double {
+        var i = 0
+        var parts = string.split('.')
+        var whole = parts[0]
+        var multiplier = 1.0
+        var out = 0.0
+        var sign = 1
+
+        // handle negative numbers
+        if (string.startsWith('-')) {
+            i = 1
+            sign = -1
+        }
+
+        fun charCodeToInt(char: Char): Int =
+            when (char) {
+                in '0'..'9' -> char - '0'
+                in 'a'..'z' -> char - 'a' + 10
+                in 'A'..'X' -> char - 'A' + 36
+                else -> throw IllegalArgumentException("Invalid character: $char")
+            }
+
+        // handle digits before the decimal
+        for (ix in i..whole.lastIndex) {
+            out = 60 * out + charCodeToInt(whole[ix])
+        }
+
+        // handle digits after the decimal
+        parts.getOrNull(1)?.let { fractional ->
+            for (c in fractional) {
+                multiplier = multiplier / 60.0
+                out += charCodeToInt(c) * multiplier
+            }
+        }
+
+        return out * sign
+    }
+
+    fun <T, R> List<T>.scanWithoutInitial(initial: R, operation: (acc: R, T) -> R): List<R> = buildList {
+        var accumulator = initial
+        for (element in this@scanWithoutInitial) {
+            accumulator = operation(accumulator, element)
+            add(accumulator)
+        }
+    }
+
+    fun List<Long>.partialSums(): List<Long> = scanWithoutInitial(0, Long::plus)
+
+    val zones = mutableMapOf<String, TimeZoneRules>()
+    val (zonesPacked, linksPacked) = readTzdb() ?: return@runCatching null
+    for (zone in zonesPacked) {
+        val components = zone.split('|')
+        val offsets = components[2].split(' ').map { unpackBase60(it) }
+        val indices = components[3].map { it - '0' }
+        val lengthsOfPeriodsWithOffsets = components[4].split(' ').map {
+            (unpackBase60(it) * SECONDS_PER_MINUTE * MILLIS_PER_ONE).roundToLong() / // minutes to milliseconds
+                    MILLIS_PER_ONE // but we only need seconds
+        }
+        zones[components[0]] = TimeZoneRules(
+            transitionEpochSeconds = lengthsOfPeriodsWithOffsets.partialSums().take<Long>(indices.size - 1),
+            offsets = indices.map { UtcOffset(null, null, -(offsets[it] * 60).roundToInt()) },
+            recurringZoneRules = null
+        )
+    }
+    for (link in linksPacked) {
+        val components = link.split('|')
+        zones[components[0]]?.let { rules ->
+            zones[components[1]] = rules
+        }
```

Comment:
Is the order of the two time zones specified? I mean is it possible that the second time zone has rules but the first doesn't?

Reply:
https://momentjs.com/timezone/docs/#/data-formats/link-format/ doesn't say anything either way, so the js-joda implementation is the best guidance we have on this, and they do assume this order.